### PR TITLE
🧹 Remove unused azure functions. Add deprecation TODOs.

### PR DESCRIPTION
**Pull request by preslavgerchev**
Small cleanup + deprecation TODOs.

### CODE REVIEW

File: motor/providers/azure/client.go [hunk 1]
```diff
@@ -18,44 +13,3 @@ func IsAzInstalled() bool {
 	command, err := t.RunCommand("az")
 	return command.ExitStatus == 0 && err == nil
 }
-
-// shells out to `az account show --output json` to determine the default account
-// call `az account list` displays all subscriptions
-func GetAccount() (*AzureAccount, error) {
```
Comment (chris-rock) [inline motor/providers/azure/client.go:24]: Shouldn't that be used by default if users type `cnquery shell azure`?

Comment (preslavgerchev) [reply motor/providers/azure/client.go:24]: The azure sdk uses `az account get-access-token -o json` to do so, we only need to check if `az` is present

Comment (chris-rock) [reply motor/providers/azure/client.go:24]: and the az command also gather the default subscription?

Comment (preslavgerchev) [reply motor/providers/azure/client.go:24]: We only use the `IsAzInstalled` to make sure that if you run `cnquery shell azure` without any params you can use the AZ cli. In that case it discovers all subs and you can pick:
image

**Comment by preslavgerchev** [comment]
Closing this in favour of #534

**Pull request closed without merging**